### PR TITLE
Feat/button enhancement

### PR DESCRIPTION
**Pull request by fredericvilcot**
This PR brings `Button` enhancement, according to following spec : https://github.com/okp4/ui/issues/119

### CODE REVIEW

**Review by amimart — APPROVED** [review]

**Review by nperon — APPROVED** [review]

File: src/ui/atoms/button/Button.tsx [hunk 1]
```diff
@@ -27,24 +37,29 @@ export type TButtonProps = Readonly<{
...
  */
 export const Button: React.FC<TButtonProps> = ({
-  primary = false,
+  variant = 'primary',
   size = 'medium',
+  backgroundColor = 'primary',
+  disabled = false,
   label,
   onClick,
   ...props
```
Comment (ccamel) [inline src/ui/atoms/button/Button.tsx:46]: Could be removed. (`props` has `{}` type as far as I see).

File: src/ui/atoms/button/Button.tsx [hunk 2]
```diff
@@ -27,24 +37,29 @@ export type TButtonProps = Readonly<{
...
+      disabled: !!disabled
     })}
+    disabled={disabled}
     onClick={onClick}
+    title={label}
     type="button"
     {...props}
   >
-    {label}
+    <Typography as="div" color="highlighted-text" fontSize="small" fontWeight="light" noWrap>
```
Comment (ccamel) [inline src/ui/atoms/button/Button.tsx:61]: From what I see, you don't handle child components, so we're not quite compliant with the spec discussed in [ui/issues/119](https://github.com/okp4/ui/issues/119#issuecomment-1100130307) regarding the capabilities to display icons in the button.

**Review by ccamel — CHANGES_REQUESTED** [review]

File: src/ui/atoms/button/Button.tsx (hunk 2 above)
Comment (fredericvilcot) [reply src/ui/atoms/button/Button.tsx:61]: @ccamel 
I updated issue to be compliant with the specification at the moment.

File: src/ui/atoms/button/Button.tsx (hunk 1 above)
Comment (fredericvilcot) [reply src/ui/atoms/button/Button.tsx:46]: Yes you are right, and I realize that this problem is present on the other atomic components and needs to be fixed.
I propose initially to delete this element and thus to partition the use of the button to the properties defined by the interface.
Subsequently, we can discuss whether we extend the type to that of the `HTMLButtonElement`

Comment (ccamel) [reply src/ui/atoms/button/Button.tsx]: Good to me!

**Review by ccamel — APPROVED** [review]

**Review by nperon — APPROVED** [review]

**Review by amimart — APPROVED** [review]

**Pull request merged**